src/dune_pkg/opam.ml

```diff
@@ -1,7 +1,52 @@
 open Stdune
 module Package_name = Dune_lang.Package_name
 
-module Repo = struct
+module Global : sig
+  val with_switch_state :
+    f:(OpamStateTypes.unlocked OpamStateTypes.switch_state -> 'a) -> 'a
+end = struct
+  let initialized = ref false
+
+  let ensure_initialized () =
+    if not !initialized then (
+      OpamSystem.init ();
+      let root = OpamStateConfig.opamroot () in
+      let (_ : OpamFile.Config.t option) =
+        OpamStateConfig.load_defaults ~lock_kind:`Lock_read root
+      in
+      OpamFormatConfig.init ();
+      OpamCoreConfig.init ~safe_mode:true ();
+      OpamRepositoryConfig.init ();
+      OpamStateConfig.init ~root_dir:root ();
+      initialized := true)
```

Comment:
Do any of the calls above modify something on the file system?

Reply:
Skimming the code it looks like the `init` functions are just setting some mutable global config records. I copied the way that opam-monorepo initializes its opam state. cc @Leonidas-from-XIV